### PR TITLE
Feature/generate report on interruption

### PR DESCRIPTION
**Pull request by clemensgrosse**
# Description

Added mechanism to generate a report in case of unexpected abortion of test execution.
A ShutDownHook triggers on JVM finish in successful and failure cases. Report generation is then triggered within the Hook, if no report has been generated before. with sending a ExecutionAbortEvent.
Flag for generated report is set once only in ExecutionEndListener, in case of successful end of test execution (ExecutionFinishEvent received by the mentioned Listener).

## Type of change

- [x ] New feature (non-breaking change which adds functionality)

# Checklist:

- [x ] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x ] New and existing tests pass locally with my changes
- [ x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by mreiche — DISMISSED** [review]

File: core/src/main/java/eu/tsystems/mms/tic/testframework/common/JVMExitHook.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package eu.tsystems.mms.tic.testframework.common;
```
Comment (martingrossmann) [inline core/src/main/java/eu/tsystems/mms/tic/testframework/common/JVMExitHook.java:1]: Please add the license text as a file header

**Review by martingrossmann — CHANGES_REQUESTED** [review]

**Comment by martingrossmann** [comment]
As discussed please add a short note in documentation about this helpful feature.

**Review by mreiche — APPROVED** [review]

File: docs/src/docs/reports/aborted-execution.adoc [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[#ExecutionAbortion]
+= Report of aborted test runs
+
+In case of unexpected abortion of a test execution the report is generated, too. . Due to abortion of still running test methods
```
Comment (mreiche) [inline docs/src/docs/reports/aborted-execution.adoc:4]: Please fix spelling "generated, too. . Due"

File: docs/src/docs/jvm/jvm-exit-hook.adoc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[#JvmExitHook]
+= JVMExitHook
+
+The `JVMExitHook` was integrated to achieve the generation of a report with Report-NG in case an <<#ExecutionAbortion, aborted test run>>. With the start of a test execution the `JVMExitHook` is added as a shutdown hook to the JVM.
+
+When the JVM is about to stop, shutdown hooks are always called by default. This includes cases of abortion, all other system errors and normal finish. In the latter case a flag indicating an already created report is set true once in the process of the report generation.
+To avoid overriding the already existing report, the `JVMExitHook` only triggers when this flag is false, as this is the only indicator of an unexpected exit and a missing report. It then sends the <<#ExecutionAbortEvent, ExecutionAbortEvent>>, which is then caught by the corresponding listeners creating a report.
```
Comment (martingrossmann) [inline docs/src/docs/jvm/jvm-exit-hook.adoc:7]: Your link to `ExecutionAbortEvent` does not work anymore because it's an own chapter now. ;-)

**Review by mreiche — DISMISSED** [review]

**Pull request merged**